modules/ssh/init.zsh

```diff
@@ -40,9 +40,19 @@ fi
 if ssh-add -l 2>&1 | grep -q 'The agent has no identities'; then
   zstyle -a ':prezto:module:ssh:load' identities '_ssh_identities'
   if (( ${#_ssh_identities} > 0 )); then
-    ssh-add "$_ssh_dir/${^_ssh_identities[@]}"  < /dev/null 2> /dev/null
+    # Check for Linux system and ssh-askpass presence
+    if [[ $(uname -s) == Linux* ]] && [[ ! -a /etc/X11/app-defaults/SshAskpass ]]; then
+      ssh-add "$_ssh_dir/${^_ssh_identities[@]}"  2> /dev/null
```

Comment:
You can probably replace `if (( ${#_ssh_identities} > 0 ))` block by inlining the check with something like:
```
ssh-add "${_ssh_identities:+$_ssh_dir/${^_ssh_identities[@]}}"  < /dev/null 2> /dev/null
```
I haven't tested though :)